docs/using-ngrok-with/k8s.md

```diff
@@ -0,0 +1,246 @@
+:::
+
+## Get started with the ngrok Ingress Controller for Kubernetes
+
+## Step 1: Get your ngrok authtoken and API key
+
+:::tip
+The ngrok Ingress Controller requires:
+- The authtoken to launch connections from ngrok's points of presence to your k8s cluster.
+- The API key to set ngrok edge configurations for your k8s cluster
+:::
+
+To started with the ngrok Ingress Controller for Kubernetes:
+
+1. Access the [ngrok Dashboard](https://dashboard.ngrok.com) with your Pro account.
+1. Click [Your Authtoken](https://dashboard.ngrok.com/get-started/your-authtoken). Copy the Authtoken to a text editor.
+1. Click [API](https://dashboard.ngrok.com/api).
+1. Follow the instructions to create a new API key. Copy the API key to a text editor.
+
+## Step 2: Setup your Kubernetes cluster and install the ngrok Ingress Controller
+
+1. In your terminal, create a k8s cluster:
+    
+    ```bash
+    k3d cluster create ngrok
+    ```
+
+1. Using `helm`, add the ngrok repo:
+
+    ```bash
+    helm repo add ngrok https://ngrok.github.io/kubernetes-ingress-controller
+    ```
+
+1. Install the ngrok Ingress Controller in your cluster, replacing `[AUTHTOKEN]` and `[API_KEY]` with your Authtoken and API key from above:
+
+    ```bash
+    export NGROK_AUTHTOKEN=[AUTHTOKEN]
+    export NGROK_API_KEY=[API_KEY]
+    helm install ngrok-ingress-controller ngrok/kubernetes-ingress-controller \
+    --set credentials.apiKey=$NGROK_API_KEY \
+    --set credentials.authtoken=$NGROK_AUTHTOKEN
+    kubectl apply -f ngrok-manifest.yaml
+    ```
+
+1. Create a manifest file (for example `ngrok-manifest.yaml`) with the following contents:
+
+    :::tip Notes:
+    - Lines 1-32: Create the 2048 app service and deployment
+    - Lines 34-51: Create the ngrok Ingress Controller. Line 42 in specific determines the ingress URL for public requests.
+    :::
+
+    ```yaml showLineNumbers
+    apiVersion: v1
+    kind: Service
+    metadata:
+    name: game-2048
+    spec:
+    ports:
+        - name: http
+        port: 80
+        targetPort: 80
+    selector:
+        app: game-2048
+    ---
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+    name: game-2048
+    spec:
+    replicas: 1
+    selector:
+        matchLabels:
+        app: game-2048
+    template:
+        metadata:
+        labels:
+            app: game-2048
+        spec:
+        containers:
+            - name: backend
+            image: alexwhen/docker-2048
+            ports:
+                - name: http
+                containerPort: 80
+    ---
+    # ngrok Ingress Controller Configuration
+    apiVersion: networking.k8s.io/v1
+    kind: Ingress
+    metadata:
+      name: game-2048-ingress
+    spec:
+      ingressClassName: ngrok
+      rules:
+        - host: my-awesome-k8s-cluster.ngrok.app
```

Comment:
might make it clear that this needs to be templated. some guides will also include a sed command to find/replace things

`sed -i "s/__PILLAR__LOCAL__DNS__/$localdns/g; s/__PILLAR__DNS__DOMAIN__/$domain/g; s/__PILLAR__DNS__SERVER__/$kubedns/g" nodelocaldns.yaml`

https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/#configuration

Reply:
Oh... I'll need some help understanding this (these pesky marketers 😄  )

Reply:
sorry, thats an overly complex example of using sed tbh 😅 
we can worry about it later. I assume in the long run we'll want to have a consistent way of people doing things like substituting values and we can work on that consistency over time